internal/executor/service/lease_requester.go

```diff
@@ -69,45 +69,51 @@ func (requester *JobLeaseRequester) LeaseJobRuns(ctx *armadacontext.Context, req
 	leaseRuns := []*executorapi.JobRunLease{}
 	runIdsToCancel := []*armadaevents.Uuid{}
 	runIdsToPreempt := []*armadaevents.Uuid{}
-	for {
-		shouldEndStreamCall := false
-		select {
-		case <-ctx.Done():
-			if ctx.Err() == context.DeadlineExceeded {
-				shouldEndStreamCall = true
-			} else {
-				return nil, ctx.Err()
-			}
-		default:
-			res, err := stream.Recv()
-			if err == io.EOF {
-				shouldEndStreamCall = true
-			} else if err != nil {
-				return nil, err
-			}
-
-			switch typed := res.GetEvent().(type) {
-			case *executorapi.LeaseStreamMessage_Lease:
-				leaseRuns = append(leaseRuns, typed.Lease)
-			case *executorapi.LeaseStreamMessage_PreemptRuns:
-				runIdsToPreempt = append(runIdsToPreempt, typed.PreemptRuns.JobRunIdsToPreempt...)
-			case *executorapi.LeaseStreamMessage_CancelRuns:
-				runIdsToCancel = append(runIdsToCancel, typed.CancelRuns.JobRunIdsToCancel...)
-			case *executorapi.LeaseStreamMessage_End:
-				shouldEndStreamCall = true
-			default:
-				log.Errorf("unexpected lease stream message type %T", typed)
-			}
+	shouldEndStream := false
+	for !shouldEndStream {
+		res, err := stream.Recv()
+		if err != nil {
+			return nil, err
 		}
 
-		if shouldEndStreamCall {
-			break
+		switch typed := res.GetEvent().(type) {
+		case *executorapi.LeaseStreamMessage_Lease:
+			leaseRuns = append(leaseRuns, typed.Lease)
+		case *executorapi.LeaseStreamMessage_PreemptRuns:
+			runIdsToPreempt = append(runIdsToPreempt, typed.PreemptRuns.JobRunIdsToPreempt...)
+		case *executorapi.LeaseStreamMessage_CancelRuns:
+			runIdsToCancel = append(runIdsToCancel, typed.CancelRuns.JobRunIdsToCancel...)
+		case *executorapi.LeaseStreamMessage_End:
+			shouldEndStream = true
+		default:
+			log.Errorf("unexpected lease stream message type %T", typed)
 		}
 	}
 
+	err = closeStream(stream)
+	if err != nil {
+		return nil, err
```

Comment:
couldn't we just log a warning here?  We've got all the information we need so do we need to propagate the error?